dnf/module/module_base.py

```diff
@@ -140,30 +140,139 @@ def install(self, module_specs, strict=True):
         if fail_safe_repo_used:
             raise dnf.exceptions.Error(_(
                 "Installing module from Fail-Safe repository is not allowed"))
-        install_base_query = self.base.sack.query().filterm(
-            nevra_strict=install_set_artefacts).apply()
+        __, profiles_errors = self._install_profiles_internal(
+            install_set_artefacts, install_dict, strict)
+        if profiles_errors:
+            error_specs.extend(profiles_errors)
 
-        # add hot-fix packages
-        hot_fix_repos = [i.id for i in self.base.repos.iter_enabled() if i.module_hotfixes]
-        hotfix_packages = self.base.sack.query().filterm(reponame=hot_fix_repos).filterm(
-            name=install_dict.keys())
-        install_base_query = install_base_query.union(hotfix_packages)
+        if no_match_specs or error_specs or solver_errors:
+            raise dnf.exceptions.MarkingErrors(no_match_group_specs=no_match_specs,
+                                               error_group_specs=error_specs,
+                                               module_depsolv_errors=solver_errors)
 
-        for pkg_name, set_specs in install_dict.items():
-            query = install_base_query.filter(name=pkg_name)
-            if not query:
-                # package can also be non-modular or part of another stream
-                query = self.base.sack.query().filterm(name=pkg_name)
-                if not query:
-                    for spec in set_specs:
-                        logger.error(_("Unable to resolve argument {}").format(spec))
-                    logger.error(_("No match for package {}").format(pkg_name))
-                    error_specs.extend(set_specs)
-                    continue
-            self.base._goal.group_members.add(pkg_name)
+    def switch(self, module_specs, strict=True):
+        # :api
+        no_match_specs, error_specs, module_dicts = self.__resolve_specs_enable(module_specs)
+        # collect name of artifacts from new modules for distrosync
+        new_artefacts_names = set()
```

Comment:
I know this is kind of stupid to comment on, but we should consistently use `artifacts` here. Swapping between `artefacts` and artifacts` is a bit dizzying...

Reply:
Good point, I will make it unoform